app/models/concerns/course/lesson_plan/item_todo_concern.rb

```diff
@@ -17,9 +17,17 @@ def can_user_start?(user)
   protected
 
   # Create todos for the given lesson_plan_item for all course_users in the course,
-  #   except invited course_users (ie. course_users who do not have a user record).
+  # except invited course_users (ie. course_users who do not have a user record).
+  #
+  # A rollback of the transaction (including saving of the actable lesson_plan_item)
+  # will be raised if the creation of the todo fails validations and cannot be created.
+  #
+  # @raise [ActiveRecord::Rollback] If the creation of the todo is invalid, the transaction will
+  #   be rolled back.
   def create_todos
     course_users = CourseUser.where(course_id: course_id).where.not(workflow_state: 'invited')
     Course::LessonPlan::Todo.create_for(self, course_users)
+  rescue ActiveRecord::RecordInvalid => error
```

Comment:
It depends on what method we call in `#create_for`. Previously it was `create`, which fails silently, and now I've changed to `create!` which would raise this error. 

#### [API Guide for `create!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence/ClassMethods.html#method-i-create-21)
> Creates an object (or multiple objects) and saves it to the database, if validations pass. Raises a RecordInvalid error if validations fail, unlike Base#create.

`create!` also uses `save!` -> do we need to rescue those as well? 
> If any of the before_* callbacks throws :abort the action is cancelled and save! raises ActiveRecord::RecordNotSaved.
[API Guide for `save!`](http://api.rubyonrails.org/classes/ActiveRecord/Persistence.html#method-i-save-21)

---

#### [Reference for `create_for`](https://github.com/Coursemology/coursemology2/blob/7fb2c21ce17ea5d98028434698f5f69d60e7cd6e/app/models/course/lesson_plan/todo.rb#L32-L45)

I'm okay to change `create_for` to return boolean, but I was hoping to follow the API for `create`. If it doesn't help us in this case, maybe returning boolean would be better - what do you think?